tests/integration_tests/tests/agent_tests/test_plugin_update.py

```diff
@@ -17,65 +17,260 @@
 import uuid
 import shutil
 
+from functools import wraps
+
 from integration_tests import AgentTestWithPlugins, BaseTestCase
-from integration_tests.tests.utils import get_resource as resource
+from integration_tests.tests.utils import \
+    (get_resource as resource,
+     wait_for_deployment_creation_to_complete)
+
+from cloudify_rest_client.exceptions import CloudifyClientError
+
+
+def setup_for_sourced_plugins(f):
+    @wraps(f)
+    def wrapper(self, *args, **kwargs):
+        self.setup_deployment_id = 'd{0}'.format(uuid.uuid4())
+        self.setup_node_id = 'node'
+        self.plugin_name = 'version-aware-plugin'
+        self.plugin_dir_name_prefix = self.plugin_name + '_'
+        self.base_name = 'base_sourced'
+        self.base_blueprint_id = 'b{0}'.format(uuid.uuid4())
+        self.mod_name = 'mod_sourced'
+        self.mod_blueprint_id = 'b{0}'.format(uuid.uuid4())
+
+        self.addCleanup(self._remove_files)
+        self._prepare_files()
+        return f(self, *args, **kwargs)
+
+    return wrapper
+
+
+def setup_for_plugins_update(f):
+    @wraps(f)
+    def wrapper(self, *args, **kwargs):
+        self.blueprint_name_prefix = 'plugins_update'
+        self.base_blueprint_id = 'b{0}'.format(uuid.uuid4())
+        _f = uploads_mock_plugins(f)
+        return _f(self, *args, **kwargs)
+
+    return wrapper
+
+
+def uploads_mock_plugins(f):
+    @wraps(f)
+    def wrapper(self, *args, **kwargs):
+        self.addCleanup(self._clear_managed_plugins)
+        return f(self, *args, **kwargs)
+
+    return wrapper
 
 
 class TestPluginUpdate(AgentTestWithPlugins):
+    versions = ['1.0', '2.0']
+    dsl_resources_path = resource(os.path.join('dsl', 'agent_tests'))
+    blueprint_name_prefix = 'plugin_update_'
+    setup_deployment_ids = None
+
+    def tearDown(self):
+        deps_to_remove = []
```

Comment:
Why is this needed? the dep id are uuid

Reply:
I think this is a mistake as well. Will verify in a bit

Reply:
It's a mistake, removed.